### PR TITLE
[Doc] Refine metadata_enable_recovery_mode docs

### PR DESCRIPTION
**Pull request by EsoragotoSpirit**
Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>

## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/sql-reference/sql-statements/Administration/ADMIN_SET_PARTITION_VERSION.md [hunk 1]
```diff
@@ -6,7 +6,9 @@ displayed_sidebar: "English"
 
 ## Description
 
-Manually setting the partition version is a dangerous operation and is only used when there is a problem with the cluster metadata. Under normal circumstances, the partition version will automatically be consistent with the tablet version.
+Sets a partition to a specific data version.
+
+Note that manually setting the partition version is a critical operation and is recommended only when problems occur in the cluster metadata. In normal circumstances, the version of a partition is automatically set consistent with that of the tablet within.
```
Comment (evelynzhaojie) [inline docs/en/sql-reference/sql-statements/Administration/ADMIN_SET_PARTITION_VERSION.md:11]: Note that manually setting the partition version is a high-risk operation

Comment (evelynzhaojie) [reply docs/en/sql-reference/sql-statements/Administration/ADMIN_SET_PARTITION_VERSION.md:11]: In normal circumstances, the version of a partition is consistent with that of the tablets within.

File: docs/en/sql-reference/sql-statements/Administration/ADMIN_SET_PARTITION_VERSION.md [hunk 2]
```diff
@@ -17,29 +19,32 @@ This operation requires the SYSTEM-level OPERATE privilege. You can follow the i
...
 
-Note:
+## Parameters
 
-1. For non-partitioned tables, partition_name is the same as the table name.
-2. For random distribution tables, partition_id needs to be used to specify the partition.
+- `table_name`: The name of the table to which the partition belongs.
+- `partition_name`: The name of the partition. You need to specify a partition using either `partition_name` or `partition_id`. For non-partitioned tables, `partition_name` is the same as the table name.
+- `partition_id`: The ID of the partition. You need to specify a partition using either `partition_name` or `partition_id`. For tables with the random bucketing strategy, you can only use `partition_id` to specify the partition.
+- `version`: The version of the partition to set.
```
Comment (evelynzhaojie) [inline docs/en/sql-reference/sql-statements/Administration/ADMIN_SET_PARTITION_VERSION.md:31]: The version you want to set for the partition.

File: docs/en/sql-reference/sql-statements/Administration/ADMIN_SET_PARTITION_VERSION.md [hunk 3]
```diff
@@ -17,29 +19,32 @@ This operation requires the SYSTEM-level OPERATE privilege. You can follow the i
...
 
-1. Set the version of table t1 to 10. t1 is a non-partitioned table.
+1. Set the version of the non-partitioned table `t1` to `10`.
 
     ```sql
     ADMIN SET TABLE t1 PARTITION(t1) TO VERSION 10;
     ```
 
-2. Set the p1 partition version of partitioned table t2 to 10
+2. Set the version of the partition `p1` from the table `t2` to `10`.
```
Comment (evelynzhaojie) [inline docs/en/sql-reference/sql-statements/Administration/ADMIN_SET_PARTITION_VERSION.md:41]: Set the version of partition `p1` in table `t2` to `10`.

**Review by evelynzhaojie — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Comment by EsoragotoSpirit** [comment]
@mergify backport branch-3.2 branch-3.1

**Pull request merged**